### ISSUE TITLE
Relative date time widget with "starting from" value is not translated properly

### ISSUE THREAD

**Issue by noahmoss** [metabase/metabase#24624]
**Describe the bug**
"From now" and "ago" in the filter widget are not translated

**To Reproduce**
1. Change your user language to anything other than English
2. Create a dashboard with a question
3. Add a date filter with the "all options" type and connect it to the question, then save the dashboard
4. Set the filter to a relative date with the "starting from" feature

You should see a mix of English and your user language. I should also note that in my example below, which uses German, if you translated "from now" or "ago" directly within the sentence it still wouldn't be grammatically correct.

**Expected behavior**
The entire filter value phrase should be translated appropriately, and make grammatical sense.

**Severity**
Makes the filter value a bit nonsensical

**Screenshots**
<img width="400" alt="image" src="https://user-images.githubusercontent.com/32746338/182948460-59b4a783-6a65-416f-87c9-65162c810569.png">

**Information about your Metabase Installation:**
Current master

**Comment by flamber** [comment]
Not sure how the frontend handles translations like this, but it looks like it is coming from
https://github.com/metabase/metabase/blob/58ea89ce2bd89d26a097bfd7518d5872c8d4839c/frontend/src/metabase/lib/query_time.js#L211-L212
And
https://github.com/metabase/metabase/blob/58ea89ce2bd89d26a097bfd7518d5872c8d4839c/frontend/src/metabase/lib/query_time.js#L377

**Comment by noahmoss** [comment]
Yeah, the main issue is that `suffix` isn't a translated string. But the additional issue is that building up a full sentence from individually-translated pieces like this is always going to produce ungrammatical nonsense in other languages.

**Comment by flamber** [comment]
@noahmoss Yeah, there's #13188/#8389 for that. It's difficult to handle translations like that. Remember Mozilla spending quite an effort to try to address this in Firefox and Thunderbird, but also made the job for translator (the humans) more complicated for those languages.

**Comment by rafpaf** [comment]
Since the portions of the codebase responsible for this issue are going to be overhauled, I will leave it unfixed for now. But here are some notes.

There's a function that helps generate strings like `5 minutes ago` and `10 years from now`. It lives in [query-time.js](https://github.com/metabase/metabase/blob/41d996bab71c36feed5bd70db7bea7ce5ccb9ac9/frontend/src/metabase-lib/queries/utils/query-time.js#L253-L272) and it looks like this:

```ts
export function formatStartingFrom(bucketing, n) {
  const suffix = n >= 0 ? "from now" : "ago";
  switch (bucketing) {
    case "minute":
      return ngettext(msgid`minute ${suffix}`, `minutes ${suffix}`, n);
    case "hour":
      return ngettext(msgid`hour ${suffix}`, `hours ${suffix}`, n);
    case "day":
      return ngettext(msgid`day ${suffix}`, `days ${suffix}`, n);
    case "week":
      return ngettext(msgid`week ${suffix}`, `weeks ${suffix}`, n);
    case "month":
      return ngettext(msgid`month ${suffix}`, `months ${suffix}`, n);
    case "quarter":
      return ngettext(msgid`quarter ${suffix}`, `quarters ${suffix}`, n);
    case "year":
      return ngettext(msgid`year ${suffix}`, `years ${suffix}`, n);
  }
  return "";
}
```

I think this leads to confusing translations, since "from now" and "ago" aren't included in the string sent to translators and so those phrases don't get translated. I'm pretty sure this is why we see text like this:

![image](https://github.com/metabase/metabase/assets/130925/30570b38-2cab-4e7a-b018-269609127a5a)

Also, we're taking the translation of "minutes from now" (with "from now" left untranslated) and prepending a number. That works in English, but not in many other languages. In French, for example, "5 hour ago" is "Il y a 5 heures".

Now, the js function above was translated, with significant alterations, into the following clojure function. It lives in [temporal_bucket.cljc](https://github.com/metabase/metabase/blob/41d996bab71c36feed5bd70db7bea7ce5ccb9ac9/src/metabase/lib/temporal_bucket.cljc#L83-L106):

```clojure
(mu/defn describe-relative-datetime :- ::lib.schema.common/non-blank-string
  "Get a translated description of a relative datetime interval, ported from
 `frontend/src/metabase-lib/queries/utils/query-time.js`.

  e.g. if the relative interval is `-1 days`, then `n` = `-1` and `unit` = `:day`.

  If `:unit` is unspecified, assume `:day`."
  [n    :- TemporalIntervalAmount
   unit :- [:maybe :keyword]]
  (let [n    (interval-n->int n)
        unit (or unit :day)]
    (cond
      (zero? n)
      (i18n/tru "Now")

      (neg? n)
      ;; this should legitimately be lowercasing in the user locale. I know system locale isn't necessarily the same
      ;; thing, but it might be. This will have to do until we have some sort of user-locale lower-case functionality
      #_ {:clj-kondo/ignore [:discouraged-var]}
      (i18n/tru "{0} {1} ago" (abs n) (str/lower-case (describe-temporal-unit (abs n) unit)))

      :else
      #_ {:clj-kondo/ignore [:discouraged-var]}
      (i18n/tru "{0} {1} from now" n (str/lower-case (describe-temporal-unit n unit))))))
```

Here, the temporal modifiers ("from now" and "ago") are treated as variables, and so are the temporal units ("minute", "year", and so on). Translators are invited to translate phrases like "{0} {1} ago", but I think it would be more helpful if they were asked to translate phrases like "{0} years ago".

cc: @iethree @ranquild (who I believe is one of the people who will be porting and improving at least some of this code in the future)

**Comment by rafpaf** [comment]
Some code that may help translate "from now" and "ago" in the FE: https://metaboat.slack.com/archives/C6UBFU9GW/p1709836623273729?thread_ts=1709657245.672539&cid=C6UBFU9GW

**Comment by Tony-metabase** [comment]
It also happens to "previous" 

<img width="1494" alt="image" src="https://github.com/user-attachments/assets/8b67cfbb-4ee3-4c7e-981f-f16404903aad">

**Comment by ranquild** [comment]
https://github.com/metabase/metabase/pull/49241 fixes the FE part of the issue. There is a BE issue that `Previous` or `Next` words are not translated correctly when we call `describe-temporal-interval`, while other parts are translated:

![Image](https://github.com/user-attachments/assets/2d647aed-1f08-4061-bb63-c9a004d82cad)
